src/param/src/lib.rs

```diff
@@ -128,26 +136,94 @@ impl Param {
 
             // Parse value.
             match key {
-                b"TITLE" => title = Some(Self::read_utf8(&mut raw, i, format, len, 128)?),
-                b"TITLE_ID" => title_id = Some(Self::read_utf8(&mut raw, i, format, len, 12)?),
+                b"APP_VER" => {
+                    app_ver = Some(Self::read_utf8(&mut raw, i, format, len, 8)?);
+                }
+                b"CATEGORY" => {
+                    let category_param = Self::read_utf8(&mut raw, i, format, 4, 4)?;
+                    category = Some(category_param.clone());
+                    if !category_param.contains("bd") && !category_param.contains("ac")  // Blu-Ray Game, DLC.
+                        // Check if this is a patch
+                        && !category_param.starts_with("gp")
+                        // Check if this is an application
+                        && !category_param.starts_with("gd")
+                    {
+                        // For types such as gc, sd, la, and wda, there is no Title or TitleID.
+                        title = Some(format!("No Title {}", category_param).to_string());
+                        title_id = Some("No TitleID".to_string());
+                    }
+                }
+                b"CONTENT_ID" => {
+                    content_id = Some(Self::read_utf8(&mut raw, i, format, len, 48)?);
+                }
+                b"TITLE" => {
+                    if title.is_none() {
+                        title = Some(Self::read_utf8(&mut raw, i, format, len, 128)?);
+                    }
+                }
+                b"TITLE_ID" => {
+                    if title_id.is_none() {
+                        title_id = Some(Self::read_utf8(&mut raw, i, format, len, 12)?);
+                    }
+                }
+                b"VERSION" => {
+                    version = Some(Self::read_utf8(&mut raw, i, format, len, 8)?);
+                }
                 _ => continue,
             }
         }
 
         Ok(Self {
+            // App_Ver for Games and Patches, for DLC, use version. Anything else is abnormal.
+            app_ver: app_ver
+                .or(version.clone())
+                .ok_or(ReadError::MissingVersion)?,
```

Comment:
Better to not set this to version if it is not available because it will hide the information from the user of this struct.

Reply:
And as for this... Uh, if you read the comment, DLC doesn't have App_Ver, and due to how you have your code... Means it'll error... Oh, I could modify it to run the app_ver check if its a game, a patch, or PS2 classic.

Reply:
You can make `app_ver` field as an `Option`. Then let the user of this struct decide what to do if this field is absent.